prime-router/src/main/kotlin/azure/ActionHistory.kt

```diff
@@ -410,6 +411,65 @@ class ActionHistory {
         )
     }
 
+    // Used as temp storage by the json generator, below.
+    private data class DestinationData(
+        val orgSvc: OrganizationService,
+        var count: Int,
+        val sendingAt: OffsetDateTime? = null,
+    )
+
+    /**
+     * Generate nice json describing the destinations, suitable for returning to a Hub client.
+     * Most of the ugliness here is the attempt to not print every 1-entry report, but combine and summarize them.
+     *
+     * This works by side-effect on jsonGen.
+     */
+    fun prettyPrintDestinationsJson(jsonGen: JsonGenerator) {
+        val metadata = WorkflowEngine.metadata
+        if (reportsOut.isEmpty()) return
+        jsonGen.writeArrayFieldStart("destinations")
+        // Avoid clutter.  Combine reports with one Item, and print combined count.
+        var singles = mutableMapOf<String, DestinationData>()
+        var destinationCounter = 0
+        reportsOut.forEach { (id, reportFile) ->
+            val fullname = reportFile.receivingOrg + "." + reportFile.receivingOrgSvc
+            val orgSvc = metadata.findService(fullname) ?: return@forEach
+            if (reportFile.itemCount == 1) {
+                var previous = singles.putIfAbsent(fullname, DestinationData(orgSvc, 0, reportFile.nextActionAt))
+                if (previous != null) previous.count++
```

Comment:
@Mr0grog , thanks for your astute comments!

1.  Currently we batch based on receiving_org_svc, so there's no loss of info in `sendingAt`.   However, you are right that we might change that grouping someday, and this would break.  I'll at a comment.
2. reports of size 1 are super common, because certain health depts can only handle one at a time submissions, so we are actually splitting submissions apart in some cases. So its totally worth treating those as a special case.   

In both cases, its worth remembering this is just decorative information back to the sender

Reply:
Got it, thanks for the explanation. 👍